include/sound/sof/topology.h

```diff
@@ -222,6 +223,9 @@ enum sof_ipc_process_type {
 	SOF_PROCESS_DEMUX,
 	SOF_PROCESS_DCBLOCK,
 	SOF_PROCESS_SMART_AMP,	/**< Smart Amplifier */
+	SOF_PROCESS_CODEC_PCM,  /**< Codecs */
+	SOF_PROCESS_CODEC_MP3,
+	SOF_PROCESS_CODEC_AAC,
```

Comment:
@dbaluta are you still going to need these with UUIDs?

Reply:
@ranj063 need to understand how UUID works. I assume that UUID is just use to select the component type. In our case the POST_PROCESS component (or as other suggested to choose a different name, maybe process or codec) can be used with multiple decoders (e.g mp3/aac) and I would expect that the "subtype" or "flavour" of the process component to be read from sof-ipc_process_type enum. Not sure if UUID can encode this for a single component. Will need to read UUID PR though.

Reply:
UUID can encode "Component MP3 decoder from Cadence" or "Component MP3 decoder from Whoever". You can generate about one UUID per ms and get a unique one so you get super specific (the only thing is that multiple versions should share UUID, but that's it).

Reply:
@paulstelian97 sure, but we need to handle this in the firmware. For example we create a new component of type process which can handle multiple types of decoders. 
Do, you say that one component can be associated with multiple UUIDs?

Reply:
Ah, no, the component itself should have one UUID (the UUID is static in the topology)

Reply:
Yes, just specify an unique UUID for your decoder module is good enough.